src/main/scala/circt/stage/CIRCTStage.scala

```diff
@@ -23,6 +24,7 @@ trait CLI { this: Shell =>
     PrintFullStackTraceAnnotation,
     ThrowOnFirstErrorAnnotation,
     WarningsAsErrorsAnnotation,
+    SourceRootAnnotation,
```

Comment:
doesn't this still need to do a translation step into the invocation to `firtool`? (sorry if you're still working on this!)

Reply:
I am now propagating it through but still need to write a test that it propagates properly, any suggestions @seldridge? Also any thoughts on if I should migrate all of the tests out of ChiselMainSpec pre-emptively since it goes away on the chisel5 branch?

Reply:
This may be difficult to test. I'm fine with _not_ testing it. There may be some magic with putting some Chisel in an alternative directory and then compiling that. I admit it's complicated. This actual feature is pretty well tested on the CIRCT side: https://github.com/llvm/circt/blob/main/test/firtool/include-dirs.fir

I moved everything that I thought made sense here: https://github.com/chipsalliance/chisel3/commit/15214885eb4354938b7310771cc90254e153cb64 Is there anything that you see that's missing?

Reply:
https://github.com/chipsalliance/chisel3/commit/15214885eb4354938b7310771cc90254e153cb64 doesn't seem to actually check that the source line and caret are there, I also added a couple of a new tests showing that multiple `--source-root` arguments are allowed and that the first wins when a similarly named source file exists in both. See the stuff I added to `src/test/resources` to help